### ISSUE TITLE
[BUG] Learning rate drops to 0 very early making the training useless

### ISSUE THREAD

**Issue by m3tze-dev** [open-edge-platform/geti_v2#1603]
### 🧩 Issue type

Bug 🐞

### 📄 Description

When training a maskRCNN-ResNet-50-V2 model or MaskRCNN-EfficientNetB2B model (possibly more) the learning rate will drop to 0 after ~30 epochs - everytime... Looks like a scheduling issue?
This makes the model training useless.
(Tried on multiple different datasets on two seperate machines running the same version of GETI)

### 📋 Steps to reproduce (if applicable)

Upload a dataset with labels
Go to Models -> Train model -> MaskRCNN-ResNet50-V2
Using default Data management and Training parameters (tried with and without early stopping, but same problem).

<img width="954" height="958" alt="Image" src="https://github.com/user-attachments/assets/10481e07-3541-4e74-9e49-e1fa31e87c30" />

### 💻 Expected behavior

I expect that the learning rate does not drop to 0 but rather go towards the base learning rate, which is default 0.007.

### 🖼️ Screenshots, video recordings or logs

[Model metrics.pdf](https://github.com/user-attachments/files/24071646/Model.metrics.pdf)

<img width="709" height="468" alt="Image" src="https://github.com/user-attachments/assets/3ec5b9f6-2c95-4c69-b6b8-7addd7929436" />

### 📦 Geti version

2.13.0

### Hardware

Intel CPU, Nvidia GPU

### 🖥️ Operating System

Ubuntu 22

**Comment by kprokofi** [comment]
Hi @m3tze-dev, for the training we use ReduceLROnPlateau with small patience (4 epochs). For target metric we use val/map_50 with a drop factor 0.1. This behavior means that metric isn't improving and LR continues to drop. To avoid training being useless we use early stopping with patience equal to 10 epochs. I can confirm that looking at the validation metric, it stops improving after 10 epochs. Severe drop happens on 17 epoch.

**Comment by m3tze-dev** [comment]
Thanks for your response.
I see from otx training extension:
    scheduler:
      class_path: otx.backend.native.schedulers.LinearWarmupSchedulerCallable
      init_args:
        num_warmup_steps: 100
        main_scheduler_callable:
          class_path: lightning.pytorch.cli.ReduceLROnPlateau
          init_args:
            mode: max
            factor: 0.1
            patience: 4
            monitor: val/map_50

I believe the values are too strict, especially for "difficult datasets".
Looks like its not possible to modify the values in GETI, so I will have to make the changes in the code myself.

**Comment by kprokofi** [comment]
> Thanks for your response. I see from otx training extension: scheduler: class_path: otx.backend.native.schedulers.LinearWarmupSchedulerCallable init_args: num_warmup_steps: 100 main_scheduler_callable: class_path: lightning.pytorch.cli.ReduceLROnPlateau init_args: mode: max factor: 0.1 patience: 4 monitor: val/map_50
> 
> I believe the values are too strict, especially for "difficult datasets". Looks like it's not possible to modify the values in GETI, so I will make the necessary code changes on my side.

I appreciate your feedback and fully agree with your observations — thank you for highlighting this.

Please keep in mind that our default configuration is designed to provide a balanced trade-off between accuracy and convergence speed, based on our internal benchmarks. In some cases, however, specific datasets may require more fine-grained adjustments.
We will review these settings and work on making them more configurable in GETI to better support such scenarios.
